byteosaurus_hex.py

```diff
@@ -1933,7 +1938,397 @@ def vxlan():
+                        except ValueError:
+                            logger.error(
+                                "Invalid input '{}' Expected comma separated values in range (0-1048575)".format(labels[k]))
+                            return None                  
+                    mpls_inputs.insert(i,labels)
+                elif "Codeword" in mpls_input_param[i] and temp_input.lower() == "y":
+                    mpls_inputs.insert(i, True)
+                elif "Codeword" in mpls_input_param[i] and temp_input.lower() == "n":
+                    mpls_inputs.insert(i, False)
+                elif "Codeword" not in mpls_input_param[i]:
+                    mpls_inputs.insert(i, temp_input)
+                else:
+                    logger.critical("Invalid choice, got '{}' expected values (y/n)".format(temp_input))
+                    return None 
+            # Common parameters
+            for j in range(0, len(common_param)):
+                i = i + 1
+                mpls_inputs.insert(i, input("{} > ".format(common_param[j])))
+            mpls_tcp_pkt = mpls_packet(fuzzy, inner_tcp_pkt, mpls_inputs)
+            if inner_tcp_pkt != None and mpls_tcp_pkt != None:
+                logger.info("Inner TCP packet built")
+                logger.info("MPLS TCP Packet built")
+                mpls_tcp_pkt.show()
+                return mpls_tcp_pkt, mpls_inputs[4], mpls_inputs[5]
+            else:
+                return None
+        else:
+            logger.critical(
+                "Invalid input '{}' Expected string (y/n)".format(fuzzy))
+            return None
+    elif msg_type == 'MPLS_ARP':
+        # Get input parameters
+        mpls_input_param, common_param = requires("MPLS")
+        arp_input_param, _ = requires('ARP')
+        inner_arp_pkt, mpls_arp_pkt = None, None
+        fuzzy = (input("Generate random MPLS ARP Packet? (y/n) > ").strip()).lower()
+        if fuzzy == 'y':
+            # Build inner ARP Packet
+            arp_type = (input("ARP Type (req/resp) > ").strip()).lower()
+            inputs = []
+            # Common parameters
+            for i in range(0, len(common_param)):
+                inputs.insert(i, input("{} > ".format(common_param[i])))
+            inner_arp_pkt = arp_packet(fuzzy, 'MPLS', arp_type, inputs)
+            # Generate MPLS packet
+            mpls_arp_pkt = mpls_packet(fuzzy, inner_arp_pkt, inputs)
+            if inner_arp_pkt != None and mpls_arp_pkt != None:
+                logger.info("Inner ARP packet built")
+                logger.info("MPLS ARP packet built")
+                mpls_arp_pkt.show()
+                return mpls_arp_pkt, inputs[0], inputs[1]
+        elif fuzzy == 'n':
+            # Build inner ARP packet
+            arp_type = (input("ARP Type (req/resp) > ").strip()).lower()
+            inputs = []
+            del arp_input_param[-1]  # Skipping VLAN tag for inner ARP
+            if arp_type == 'req':
```

Comment:
Need to remove this section and replace this with following

```
--- a/byteosaurus_hex.py
+++ b/byteosaurus_hex.py
@@ -2250,10 +2250,10 @@ def build_mpls(msg_type):
             arp_type = (input("ARP Type (req/resp) > ").strip()).lower()
             inputs = []
             del arp_input_param[-1]  # Skipping VLAN tag for inner ARP
-            if arp_type == 'req':
-                del arp_input_param[-2] #skipping Target MAC
-                del arp_input_param[-4] #skipping Destination MAC - These are known for ARP request
             for i in range(0, len(arp_input_param)):
+                if arp_type == 'req' and (arp_input_param[i] in ['Destination MAC', 'Target MAC']): #skipping Target & Destination MAC as known for ARP Request
+                    inputs.insert(i, None)
+                    continue
                 temp_input = input("Inner {} > ".format(arp_input_param[i]))
                 inputs.insert(i, temp_input)
             inner_arp_pkt = arp_packet(fuzzy, 'MPLS', arp_type, inputs)
```

As currently if we remove those inputs, the code breaks 

```shell
Building flow number [ 1 ]:

Packet Type:

1 -- MPLS - Inner ICMP
2 -- MPLS - Inner UDP
3 -- MPLS - Inner TCP
4 -- MPLS - Inner ARP

Enter your choice (1-4) > 4
Generate random MPLS ARP Packet? (y/n) > n
ARP Type (req/resp) > req
Inner Source MAC (de:ad:be:ef:ca:fe) > f5:de:a9:56:b1:c9
Inner Sender MAC > f5:de:a9:56:b1:c9
Inner Sender IP > 100.100.100.1
Inner Target IP > 100.100.100.2
Stack Summary: /mnt/flash/scapy-2.5.0/byteosaurus_hex.py:<module> /mnt/flash/scapy-2.5.0/byteosaurus_hex.py:callModule /mnt/flash/scapy-2.5.0/byteosaurus_hex.py:mpls /mnt/flash/scapy-2.5.0/byteosaurus_hex.py:build_mpls /mnt/flash/scapy-2.5.0/byteosaurus_hex.py:arp_packet
IndexError: list index out of range
Traceback (most recent call last):
  File "/mnt/flash/scapy-2.5.0/byteosaurus_hex.py", line 2653, in <module>
    callModule(call_mod)
  File "/mnt/flash/scapy-2.5.0/byteosaurus_hex.py", line 2573, in callModule
    flow_instance = mpls()
  File "/mnt/flash/scapy-2.5.0/byteosaurus_hex.py", line 2328, in mpls
    return build_mpls("MPLS_ARP")
  File "/mnt/flash/scapy-2.5.0/byteosaurus_hex.py", line 2262, in build_mpls
    inner_arp_pkt = arp_packet(fuzzy, 'MPLS', arp_type, inputs)
  File "/mnt/flash/scapy-2.5.0/byteosaurus_hex.py", line 1468, in arp_packet
    trgt_ip = arp_inputs[5]
IndexError: list index out of range
```

This is because in `arp_packet` we are reusing the inputs for both arp_req and arp_resp and then overriding based on type

```python
    elif fuzzy == 'n':
        if module != None and len(arp_inputs) != 0:
            if module == 'ARP' or module == 'VXLAN' or module == 'MPLS':
                src_mac = arp_inputs[0]
                sender_mac = arp_inputs[2]
                sender_ip = arp_inputs[3]
                trgt_ip = arp_inputs[5]
                if arp_type == "req":
                    dst_mac = 'ff:ff:ff:ff:ff:ff'
                    trgt_mac = '00:00:00:00:00:00'
                elif arp_type == 'resp':
                    dst_mac = arp_inputs[1]
                    trgt_mac = arp_inputs[4]
```

So basically this is a menu bug, where we are still asking user to input Destination and target Mac even though we are setting those statically in code.

```shell
1 -- ICMP
2 -- ARP
3 -- IGMP
4 -- Multicast
5 -- VXLAN
6 -- Pause Frame
7 -- Priority Flow Control
8 -- MPLS
9 -- Load PCAP File
10 -- Exit

Enter your choice (1-10): 2
Enter the number of flows > 1

Building flow number [ 1 ]:

Generate random ARP Packet? (y/n) > n
ARP Type (req/resp) > req
Source MAC (de:ad:be:ef:ca:fe) > f5:de:a9:56:b1:c9
Destination MAC >
Sender MAC > f5:de:a9:56:b1:c9
Sender IP > 1.1.1.1
Target MAC >
Target IP > 2.2.2.2
Tag (y/n) > y
VLAN Tag (x,y) > 10,20
CoS (x,y | default 0) > 5,5
Count (c for continous) > 5
Source Interface > et3_1
2024-04-23 11:46:18,972: INFO: ARP packet built
###[ Ethernet ]###
  dst       = ff:ff:ff:ff:ff:ff
  src       = f5:de:a9:56:b1:c9
  type      = n_802_1Q
###[ 802.1Q ]###
     prio      = 5
     id        = 0
     vlan      = 10
     type      = n_802_1Q
###[ 802.1Q ]###
        prio      = 5
        id        = 0
        vlan      = 20
        type      = ARP
###[ ARP ]###
           hwtype    = Ethernet (10Mb)
           ptype     = IPv4
           hwlen     = None
           plen      = None
           op        = who-has
           hwsrc     = f5:de:a9:56:b1:c9
           psrc      = 1.1.1.1
           hwdst     = 00:00:00:00:00:00
           pdst      = 2.2.2.2
```

Once this is merged I will open a new PR to fix the inputs for the arp and VXLAN modules.